tests/codegen-llvm/issues/vec-clear-unnecessary-branch-45459.rs

```diff
@@ -0,0 +1,14 @@
+// Tests that compare and branch are optimized out when clearing a `Vec`, fixed
+// since  1.30.
+
+//@ compile-flags: -Copt-level=3
+
+#![crate_type = "lib"]
+
+// CHECK-LABEL: @check_no_compare
+#[no_mangle]
+pub fn check_no_compare(v: &mut Vec<f32>) {
+    // CHECK-NOT: icmp
+    // CHECK-NOT: br
```

Comment:
This also seems brittle -- I could see a future compiler making this just `call Vec::clear` (given that the code sequence is the same for all 4-byte vec values, and maybe even for all T: Copy vec values?).